File: src/d1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

int main()
{
    ifstream f;
    f.open("../i/i1.txt");
    string l;
    vector<string> lines;
    while (f >> l)
    {
        lines.push_back(l);
    }

    int count = 0;
    int prev = stoi(lines[0]);
    for (string& l : lines){
        int cur = stoi(l);
        if (cur > prev) {
            ++count;
        }
        prev = cur;
    }
    cout << "p1 : " << count << endl;

    count = 0;
    int old_sliding = stoi(lines[0]) + stoi(lines[1]) + stoi(lines[2]);
    for (int i = 3; i < lines.size(); ++i) {
        int cur = old_sliding + stoi(lines[i]) - stoi(lines[i - 3]);
        if (cur > old_sliding){
            ++count;
        }
        old_sliding = cur;
    }
    cout << "p2 : " << count << endl;
    
    return 0;
}

File: src/d1.py
import os

with open(os.path.join(os.getcwd(), 'i', 'i1.txt'), 'r') as f:
    lines = f.readlines()

count = 0
prev = int(lines[0])
for l in lines[1:]:
    cur = int(l)
    if (cur > prev):
        count += 1
    prev = cur
print('p1', count)

count = 0
old_sliding = int(lines[0]) + int(lines[1]) + int(lines[2])
for i, l in enumerate(lines[3:]):
    cur = old_sliding + int(l) - int(lines[i])
    if cur > old_sliding:
        count +=1
    old_sliding = cur
print('p2', count)